### PR TITLE
fix: update encryptLocalMessage and decryptLocalMessage methods to accept nonce instead of project ID

### LINKED ISSUE #17: update `nonceFromProjectId` to account for project ID creation via `projectKeyToPublicId`
Getting this error when trying to call `keyManager.encryptLocalMessage()` and passing in a `projectId` created via `projectKeyToPublicId`:

```sh
/node_modules/@mapeo/crypto/key-manager.js:127
    sodium.crypto_aead_xchacha20poly1305_ietf_encrypt(
           ^

Error: "npub" must be crypto_aead_xchacha20poly1305_ietf_NPUBBYTES bytes long
```

Both `encryptLocalMessage` and `decryptLocalMessage` use `nonceFromProjectId`, which still assumes that the project id is the hex string representation of the project key:

https://github.com/digidem/mapeo-crypto/blob/63b58907fcf7c775475c9ae0650607defe081560/key-manager.js#L225-L231

### PR DESCRIPTION
**Pull request by achou11**
Fixes #17 

~See https://github.com/digidem/mapeo-crypto/commit/40511a0e0698fdcfcda467ac89770097ee8c26c6 for changes relevant to fix. Second commit just contains prettier-formatting applied to changed files.~ EDIT: No longer relevant, as fix changed completely and formatting changes are omitted from PR

### CODE REVIEW

File: key-manager.js [hunk 1]
```diff
@@ -222,10 +223,12 @@ class KeyManager {
...
 module.exports = KeyManager
 
-/** @param {string} projectId */
-function nonceFromProjectId (projectId) {
-  return Buffer.from(projectId, 'hex').subarray(
-    0,
-    sodium.crypto_aead_xchacha20poly1305_ietf_NPUBBYTES
-  )
+/** @param {string} projectId z-base-32 encoded hash of the project key */
+function nonceFromProjectId(projectId) {
```
Comment (achou11) [inline key-manager.js:227]: looking at this, I'm wondering if this should be more flexible as opposed to assuming the usage of the `projectKeyToPublicId` util. Like what if the consumer creates a project id using some other means?

**Comment by gmaclennan** [comment]
I'm wondering that maybe it makes sense to just change the arg here to be a nonce eg a buffer of nonce bytes long? Rather than hard-coding knowledge of the project key into @mapeo/crypto what do you think?

**Comment by achou11** [comment]
> I'm wondering that maybe it makes sense to just change the arg here to be a nonce eg a buffer of nonce bytes long? Rather than hard-coding knowledge of the project key into @mapeo/crypto what do you think?

Yeah was thinking along similar lines after my initial comment! Let the consumer decide what the nonce is 😄

**Comment by achou11** [comment]
wasn't sure if running `npm run docs` is the proper way to update the API docs. seemed to give me a rather large diff so I updated the docs manually in https://github.com/digidem/mapeo-crypto/pull/18/commits/20fbec3ee00512f83acd51d103f33a696ee77897

File: key-manager.js [hunk 2]
```diff
@@ -89,13 +90,12 @@ class KeyManager {
   }
 
   /**
-   * Decrypt an encrypted message (uses the first 24 bytes of the projectId as a nonce)
+   * Decrypt an encrypted message using the provided nonce parameter
    *
    * @param {Buffer} cyphertext
-   * @param {string} projectId
+   * @param {Buffer} nonce Buffer of length sodium.crypto_aead_xchacha20poly1305_ietf_NPUBBYTES bytes
```
Comment (achou11) [inline key-manager.js:96]: is it clearer to just say the exact length (24 bytes), assuming it doesn't change?

File: key-manager.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // @ts-check
 const sodium = require('sodium-native')
 const assert = require('assert')
+const z32 = require('z32')
```
Comment (gmaclennan) [inline key-manager.js:4]: I don't think this is needed now?
```suggestion
```

**Review by gmaclennan — CHANGES_REQUESTED** [review]

**Comment by gmaclennan** [comment]
> wasn't sure if running `npm run docs` is the proper way to update the API docs. seemed to give me a rather large diff so I updated the docs manually in [6b3ac8c](https://github.com/digidem/mapeo-crypto/commit/6b3ac8cad83fc2ed67ef36ed600bbade6e0496bd)

Yeah I remember I couldn't get it to format the docs how I wanted, so I generally run it then copy and paste into the actual readme manually

**Review by gmaclennan — APPROVED** [review]

**Pull request merged**